### PR TITLE
Suggest increasing the severity for bugs with P1 WebCompat priority

### LINKED ISSUE #1408: Suggest increasing severity for bugs with WebCompat Priority set to P1

### PR DESCRIPTION
**Pull request by suhaibmujahid**
Resolves #1408

## Autonag wiki page
```wiki
{{AutonagRule
  | Rule name = Bugs with P1 WebCompat priority and severity set to low
  | Purpose   = Consistency and getting traction on important web compatibility bugs
  | Action    = Needinfo the assignee or the triage owner if not assigned yet
  | Source    = severity_high_compat_priority.py 
}}
```

## Dry-run

```
2022-05-11 13:44:19,159 - INFO - Run tool severity_high_compat_priority.py
2022-05-11 13:44:23,909 - INFO - Tool severity_high_compat_priority.py has finished.
```

<details>
<summary>Click to expand the dry-run output!</summary>

<br>
<p>The following bugs have a P1 WebCompat priority, however, their severity is set to low. Owners got needinfo requests to bump the severity:
<table style="border:1px solid black;border-collapse:collapse;" border="1">
<thead>
<tr>
<th>Bug</th><th>Summary</th><th>Severity</th>
</tr>
</thead>
<tbody>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1752655">1752655</a>
</td>
<td>
Slack issues on session restore: custom font not loaded, sidebar with thread conversation broken or: &#34;For some reason, Slack couldn&#8217;t load&#34;
</td>
<td>
S3
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1680999">1680999</a>
</td>
<td>
www.bose.com - Instant  browser crash
</td>
<td>
S3
</td>
</tr>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1651588">1651588</a>
</td>
<td>
Text deleted in contenteditable=&#34;true&#34; element on drag and drop
</td>
<td>
S3
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1615852">1615852</a>
</td>
<td>
broken `&lt;br&gt;` workaround in `&lt;div contenteditable=&#34;true&#34;&gt;` with bad side effects (also affects Google Sheets)
</td>
<td>
S3
</td>
</tr>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1530370">1530370</a>
</td>
<td>
[meta] Support CTAP2 (FIDO2) Passwordless Web Authentication
</td>
<td>
normal
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1524772">1524772</a>
</td>
<td>
Webcompat: Offline unavailable in Gmail
</td>
<td>
normal
</td>
</tr>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1520297">1520297</a>
</td>
<td>
Preload Intermediates on Android
</td>
<td>
normal
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1498512">1498512</a>
</td>
<td>
Enable by default setSinkId pref
</td>
<td>
normal
</td>
</tr>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1423746">1423746</a>
</td>
<td>
CSS &#39;filter&#39; on the root element (documentElement) should not establish a containing block for absolute/fixed positioned descendants
</td>
<td>
normal
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1123938">1123938</a>
</td>
<td>
[meta] Make viewport handling consistent with other browsers
</td>
<td>
normal
</td>
</tr>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=939897">939897</a>
</td>
<td>
[meta] Support shifting / splitting flex items between continuations of a fragmented flex container (flexbox content truncated when printing)
</td>
<td>
S3
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=390936">390936</a>
</td>
<td>
Implement Internet Explorer zoom property for css
</td>
<td>
normal
</td>
</tr>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=306344">306344</a>
</td>
<td>
Marquee should be able to use smoothscroll code
</td>
<td>
normal
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=92921">92921</a>
</td>
<td>
contenteditable should not replace linebreaks by &lt;br&gt; inside a &lt;pre&gt;
</td>
<td>
S3
</td>
</tr>
</tbody>
</table>
</p>
<p>Sincerely,<br>
Autonag bot
</p>
<p>
See the search query on <a href="https://mzl.la/37zRoNo">Bugzilla</a>.
</p>

<p>
See the source / report issues on the <a href="https://github.com/mozilla/relman-auto-nag/">GitHub repo</a>
</p>
</details>

### CODE REVIEW

**Comment by marco-c** [comment]
We should use the "WebCompat Priority" field rather than the component, e.g. https://bugzilla.mozilla.org/buglist.cgi?v1=P1&o1=equals&query_format=advanced&resolution=---&f1=cf_webcompat_priority&classification=Client%20Software&classification=Developer%20Infrastructure&classification=Components&classification=Server%20Software&classification=Other&list_id=16080400.

**Review by marco-c — CHANGES_REQUESTED** [review]
Marking as changes requested to get this out of my review queue :)

File: auto_nag/scripts/severity_high_compat_priority.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                return {"mail": person, "nickname": bug[f"{field}_detail"]["nick"]}
+
+        return None
+
+    def get_bz_params(self, date):
+        fields = ["triage_owner", "assigned_to", "severity"]
+
+        params = {
+            "include_fields": fields,
+            "resolution": "---",
```
Comment (marco-c) [inline auto_nag/scripts/severity_high_compat_priority.py:44]: Let's only do this for bugs of type `defect`, as severity doesn't apply to other types.

Comment (suhaibmujahid) [reply auto_nag/scripts/severity_high_compat_priority.py:44]: Make sense. Done in 27c1cf2410aad911cd5f3fece345eb7d418875d4




<details>
<summary>Click to expand the dry-run output!</summary>

<br>

<p>The following bugs have a P1 WebCompat priority, however, their severity is set to low. Owners got needinfo requests to bump the severity:
<table style="border:1px solid black;border-collapse:collapse;" border="1">
<thead>
<tr>
<th>Bug</th><th>Summary</th><th>Severity</th>
</tr>
</thead>
<tbody>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1752655">1752655</a>
</td>
<td>
Slack issues on session restore: custom font not loaded, sidebar with thread conversation broken or: &#34;For some reason, Slack couldn&#8217;t load&#34;
</td>
<td>
S3
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1680999">1680999</a>
</td>
<td>
www.bose.com - Instant  browser crash
</td>
<td>
S3
</td>
</tr>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1651588">1651588</a>
</td>
<td>
Text deleted in contenteditable=&#34;true&#34; element on drag and drop
</td>
<td>
S3
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1615852">1615852</a>
</td>
<td>
broken `&lt;br&gt;` workaround in `&lt;div contenteditable=&#34;true&#34;&gt;` with bad side effects (also affects Google Sheets)
</td>
<td>
S3
</td>
</tr>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1524772">1524772</a>
</td>
<td>
Webcompat: Offline unavailable in Gmail
</td>
<td>
normal
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1520297">1520297</a>
</td>
<td>
Preload Intermediates on Android
</td>
<td>
normal
</td>
</tr>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1423746">1423746</a>
</td>
<td>
CSS &#39;filter&#39; on the root element (documentElement) should not establish a containing block for absolute/fixed positioned descendants
</td>
<td>
normal
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1123938">1123938</a>
</td>
<td>
[meta] Make viewport handling consistent with other browsers
</td>
<td>
normal
</td>
</tr>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=92921">92921</a>
</td>
<td>
contenteditable should not replace linebreaks by &lt;br&gt; inside a &lt;pre&gt;
</td>
<td>
S3
</td>
</tr>
</tbody>
</table>
</p>
<p>Sincerely,<br>
Autonag bot
</p>
<p>
See the search query on <a href="https://mzl.la/3ytRuB8">Bugzilla</a>.
</p>

<p>
See the source / report issues on the <a href="https://github.com/mozilla/relman-auto-nag/">GitHub repo</a>
</p>

</details>

**Review by marco-c — APPROVED** [review]

**Pull request merged**